homework/index.js

```diff
@@ -30,14 +30,129 @@
     return elem;
   }
 
+  function createOptions(repositories, select) {
+    repositories.forEach((repository, index) => {
+      createAndAppend('option', select, {
+        text: repository.name,
+        value: index,
+      });
+    });
+  }
+
+  function addRow(repository, table, property) {
+    const tbody = createAndAppend('tbody', table);
+    const tr = createAndAppend('tr', tbody);
+    createAndAppend('th', tr, {
+      text: `${property.toUpperCase()}: `,
+    });
+    if (property !== 'name') {
+      createAndAppend('td', tr, {
+        text: `${repository[property]}`,
+        id: property,
+      });
+    } else {
+      const firstRow = createAndAppend('td', tr);
+      createAndAppend('a', firstRow, {
+        text: `${repository[property]}`,
+        href: repository.html_url,
+      });
+    }
+  }
+
+  function renderRepositoryBasicInfo(repository, table) {
+    table.innerHTML = '';
+    addRow(repository, table, 'name');
+    addRow(repository, table, 'description');
+    addRow(repository, table, 'forks');
+    addRow(repository, table, 'updated_at');
+    const upDate = document.getElementById('updated_at');
+    upDate.innerHTML = '';
+    const dateArray = repository.updated_at.split('T');
+    upDate.innerHTML = `<em>${dateArray[0]}</em>, ${dateArray[1].substring(
+      0,
+      dateArray[1].length - 1,
+    )}`;
+  }
+
+  function renderError(err) {
+    const root = document.getElementById('root');
+    createAndAppend('div', root, {
+      text: err.message,
+      class: 'alert-error',
+    });
+  }
+
+  function renderContributions(repository, ul) {
+    const url = repository.contributors_url;
+    fetchJSON(url, (err, contributions) => {
+      if (err) {
+        renderError(err);
+        return;
+      }
+      ul.innerHTML = '';
+      contributions.forEach(contributor => {
+        const li = createAndAppend('li', ul);
+        const section = createAndAppend('section', li);
+        createAndAppend('img', section, {
+          src: contributor.avatar_url,
+          alt: contributor.login,
+        });
+        const sectionName = createAndAppend('section', li);
+        const h3 = createAndAppend('h3', sectionName);
+        createAndAppend('a', h3, {
+          text: contributor.login,
+          href: contributor.html_url,
+        });
+        createAndAppend('h4', sectionName, {
+          text: contributor.contributions,
+          class: 'num',
+        });
+      });
+    });
+  }
+
   function main(url) {
     fetchJSON(url, (err, repositories) => {
       const root = document.getElementById('root');
       if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+        renderError(err);
         return;
       }
-      createAndAppend('pre', root, { text: JSON.stringify(repositories, null, 2) });
+      // create header
```

Comment:
I can already conclude that you are creating a reader by reading the code. What is the added value of this comment?